test/relayFeeCalculator.test.ts

```diff
@@ -272,3 +290,73 @@ describe("RelayFeeCalculator", () => {
     assert.equal(client.capitalFeePercent("0", "WBTC").toString(), Number.MAX_SAFE_INTEGER.toString());
   });
 });
+
+describe("RelayFeeCalculator: Composable Bridging", function () {
+  let spokePool: Contract, erc20: Contract, destErc20: Contract, weth: Contract;
+  let client: RelayFeeCalculator;
+  let queries: EthereumQueries;
+  let testContract: Contract;
+  let owner: SignerWithAddress, relayer: SignerWithAddress;
+  let tokenMap: typeof TOKEN_SYMBOLS_MAP;
+  let testGasFeePct: (message?: string) => Promise<BigNumber>;
+
+  beforeEach(async function () {
+    [owner, relayer] = await ethers.getSigners();
+
+    ({ spokePool, erc20, weth, destErc20 } = await deploySpokePoolWithToken(1, 10));
+
+    (tokenMap = {
+      USDC: {
+        name: "USDC",
+        symbol: "USDC",
+        decimals: 6,
+        addresses: {
+          1: erc20.address,
+          10: erc20.address,
+        },
+      },
+    } as unknown as typeof TOKEN_SYMBOLS_MAP),
+      await spokePool.setChainId(10); // The spoke pool for a fill should be at the destinationChainId.
+    await setupTokensForWallet(spokePool, relayer, [erc20, destErc20], weth, 100);
+
+    testContract = await hre["upgrades"].deployProxy(await getContractFactory("MockAcrossMessageContract", owner), []);
+    queries = new EthereumQueries(spokePool.provider, tokenMap, spokePool.address, relayer.address);
+    client = new RelayFeeCalculator({ queries, capitalCostsConfig: testCapitalCostsConfig });
+
+    testGasFeePct = (message?: string) =>
+      client.gasFeePercent(
+        {
+          amount: bnOne,
+          quoteTimestamp: 1,
+          recipient: testContract.address,
+          relayerFeePct: bnZero,
+          depositId: 1000000,
+          depositor: randomAddress(),
+          originChainId: 1,
+          destinationChainId: 10,
+          originToken: erc20.address,
+          destinationToken: erc20.address,
+          message: message || EMPTY_MESSAGE,
+          realizedLpFeePct: bnZero,
+        },
+        1,
+        false,
+        relayer.address,
+        1,
+        tokenMap
+      );
+  });
+  it("should not revert if no message is passed", async () => {
+    await assertPromisePasses(testGasFeePct());
+  });
+  it("should revert if the contract message fails", async () => {
+    // Per our test contract, a single byte message of 0x02 will revert.
+    const message = "0x02";
+    await assertPromiseError(testGasFeePct(message), "MockAcrossMessageContract: revert");
+  });
+  it("should be more gas to call a contract with a message", async () => {
```

Comment:
Can we be more precise? Can we get tighter bounds around the expected gas expenditure to test that the relayFeeCalculator correctly adds the gas cost to run the message to the total gas fee?

Reply:
Good call - added further bounds